pkg/rungroup/rungroup.go

```diff
@@ -65,15 +73,39 @@ func (g *Group) Run() error {
 	level.Debug(g.logger).Log("msg", "received interrupt error from first actor -- shutting down other actors", "err", initialActorErr)
 
 	// Signal all actors to stop.
+	numActors := int64(len(g.actors))
+	interruptWait := semaphore.NewWeighted(numActors)
```

Comment:
`semaphore.NewWeighted` is interesting as compared to a waitgroup. I think it's good, though a little harder to think about all the edges. 

Maybe cleaner than the channels I used for https://github.com/osquery/osquery-go/pull/108, less sure about perf. But that's not a real concern here.